src/Builders/MessageBuilder.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Builders;
+
+use InvalidArgumentException;
+use WordPress\AiClient\Files\DTO\File;
+use WordPress\AiClient\Messages\DTO\Message;
+use WordPress\AiClient\Messages\DTO\MessagePart;
+use WordPress\AiClient\Messages\Enums\MessageRoleEnum;
+use WordPress\AiClient\Tools\DTO\FunctionCall;
+use WordPress\AiClient\Tools\DTO\FunctionResponse;
+
+/**
+ * Fluent builder for constructing AI messages.
+ *
+ * This class provides a fluent interface for building messages with various
+ * content types including text, files, function calls, and function responses.
+ *
+ * @since n.e.x.t
+ */
+class MessageBuilder
+{
+    /**
+     * @var MessageRoleEnum|null The role of the message sender.
+     */
+    protected ?MessageRoleEnum $role = null;
+
+    /**
+     * @var list<MessagePart> The parts that make up the message.
+     */
+    protected array $parts = [];
+
+    /**
+     * Constructor.
+     *
+     * @since n.e.x.t
+     *
+     * @param string|null $text Optional initial text content.
+     * @param MessageRoleEnum|null $role Optional role.
+     */
+    public function __construct(?string $text = null, ?MessageRoleEnum $role = null)
```

Comment:
This should allow more than just `$text` - let's make it `mixed $input`, similar to `PromptBuilder`.

We should add parity here for all possibilities that identify something _within_ a `Message`, at least anything that can become a single `MessagePart`.

At least this should support passing a `string` (text) or a `MessagePart`, or a `MessagePartArrayShape`. Maybe even the lower-level pieces to turn into a `MessagePart` (which `string` is already one of), like `File`, `FunctionCall`, or `FunctionResponse`.